scripts/merge-analyses.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python
+
+# This script merges the analysis data in the files provided as
+# arguments, and prints the merged analysis data to stdout.
+# The "target" data lines from the input files are copied to
+# the output as-is. The "source" data lines are merged such
+# that the syntax and symbol properties are unioned across
+# all input lines that have a matching (loc, pretty) tuple.
+# This ensures that for a given identifier, only a single context
+# menu item will be displayed for a given "pretty" representation,
+# and that context menu will link to all the symbols from all
+# the input files that match that.
+
+# Note that this code intentionally preserves the order of stuff
+# within each line, as much as possible. However, the lines
+# themselves get reordered, but the output can be piped through
+# `sort` as needed.
+
+from collections import OrderedDict
+import json
+import sys
+
+sourcemap = {}
+
+def SplitToOrderedSet(comma_sep_str):
+    return OrderedDict([(item, None) for item in comma_sep_str.split(',')])
+
+for filename in sys.argv[1:]:
+    filehandle = open(filename)
+    for line in filehandle.readlines():
+        # parse the JSON into an OrderedDict to maintain order of properties
+        obj = json.loads(line, object_pairs_hook=OrderedDict)
+
+        if "source" not in obj:
+            print(line.strip())
+            continue
+
+        key = obj["loc"] + obj["pretty"]
```

Comment:
I think it would be better if you used (obj["loc"], obj["pretty"]) as your key, to avoid concerns about collisions between the ambiguity of the end of loc and the start of pretty (eg loc = FooBar and pretty = Baz would collide with loc = Foo and pretty = BarBaz.) Technically it might be a little more efficient, too, because you aren't copying the strings. IIRC tuple types in Python are immutable and can be used as keys in dicts.

Reply:
Makes sense, will do.